common/ledger/blkstorage/blockindex.go

```diff
@@ -326,6 +336,47 @@ func (index *blockIndex) exportUniqueTxIDs(dir string, newHashFunc snapshot.NewH
 	}, nil
 }
 
+func importTxIDsFromSnapshot(
+	snapshotDir string,
+	lastBlockNumInSnapshot uint64,
+	db *leveldbhelper.DBHandle) error {
+
+	batch := leveldbhelper.NewUpdateBatch()
+	txIDsMetadata, err := snapshot.OpenFile(filepath.Join(snapshotDir, snapshotMetadataFileName), snapshotFileFormat)
+	if err != nil {
+		return err
+	}
+	numTxIDs, err := txIDsMetadata.DecodeUVarInt()
+	if err != nil {
+		return err
+	}
+	txIDsData, err := snapshot.OpenFile(filepath.Join(snapshotDir, snapshotDataFileName), snapshotFileFormat)
+	if err != nil {
+		return err
+	}
+	for i := uint64(0); i < numTxIDs; i++ {
+		txID, err := txIDsData.DecodeString()
+		if err != nil {
+			return err
+		}
+		batch.Put(
+			constructTxIDKey(txID, lastBlockNumInSnapshot, uint64(i)),
```

Comment:
I too went through this process and may be you missed that I discussed this before, the only use I could think of was to improved error message for txID queries, so we decided to keep it to minimum. Yes, we still have time to change this if we want.

Reply:
I see. If it was discussed in the scrum, there is a high chance that I missed it due to procrastination. I agree that it can be fixed later if required. As I thought about this only now, I just wanted to hear your analysis.